### PR TITLE
CircleCI 2 config

### PR DESCRIPTION
**Pull request by nebril**
All basic test jobs are running in parallel, there could be some improvements here and there, but it works.

Not sure about deployment job, @adamreese @thomastaylor312 PTAL.

### CODE REVIEW

**Comment by thomastaylor312** [comment]
I'll take a look at this tonight. I'll give the deployment stuff a look too, but I am betting that @adamreese knows the most

File: .circleci/config.yml
Comment (adamreese) [inline .circleci/config.yml]: The installing go steps can be removed because it is running in the `golang:1.8` image

File: .circleci/config.yml
Comment (adamreese) [inline .circleci/config.yml]: Can all this path munging be removed now?

Comment (thomastaylor312) [inline .circleci/config.yml]: I had to manually add the Go bin in last time I did a Circle 2 file

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: Likewise here, the go installation steps can be pulled out

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: We won't need `WORKDIR` any more. We can now set the working directory with `working_directory`. See an example [here](https://github.com/thomastaylor312/helm-commandeer/blob/master/.circleci/config.yml#L4)

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: You won't need this with `working_directory` specified

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: Won't need the `cd` any more with `working_directory`

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: I would cache the dependencies after they are installed and restore the cache in future build steps (unless we end up using workflows). See the docs [here](https://circleci.com/docs/2.0/configuration-reference/#save_cache)

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: Obviously out of scope for this PR, but I wanted to capture somewhere that we should probably use `workflows` in the future to make some of these jobs (not steps) run at the same time

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: I am not sure if this is going to have the behavior we want. It is going to try and run each step of this job in parallel. See the docs [here](https://circleci.com/docs/2.0/parallelism-faster-jobs/) and [here](https://circleci.com/docs/2.0/configuration-reference/#build).

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Some changes needed, but this looks pretty good. If you have any questions about my comments please let me know 🙂

File: .circleci/config.yml
Comment (thomastaylor312) [inline .circleci/config.yml]: To clarify, I worry about it trying to run the tests and `make bootstrap` at the same time. Have you run into those issues?

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: True.

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: Each parallel build instance has it's own container, so it should be fine.

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: True, will fix

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: Will do

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: Will fix

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: Didn't find these docs, thanks!

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: kk, thanks

**Comment by nebril** [comment]
@thomastaylor312 thanks for the feedback, fixes resulted in much smaller config file

File: .circleci/config.yml
Comment (nebril) [inline .circleci/config.yml]: Makes sense, we can track it in an issue after this PR is merged

**Review by thomastaylor312 — APPROVED** [review]
This is looking great! Thanks @nebril! I'll wait for @adamreese to sign off too because he started the review

**Comment by adamreese** [comment]
Let merge after the 2.6.1 release

**Comment by adamreese** [comment]
@nebril Feel free to merge this, just ping me when you do

**Pull request merged**
